snappi_ixnetwork/ixnetworkapi.py

```diff
@@ -210,14 +210,19 @@ def get_metrics(self, request):
         # Need to change the code style when the choice Enum grows big
         if request.choice == 'port':
             response = self.vport.results(request.port)
-            return self.metrics_response().port_metrics.\
-                deserialize(response)
+            metric_res = self.metrics_response()
+            metric_res.port_metrics.deserialize(response)
+            return metric_res
```

Comment:
better use elif as we are decide according to choice

Reply:
true but, choice can hold anyone of the metric right. 
from my thought elif would be needed when the choice holds multiple values. correct me if I am wrong

Reply:
current code may not handle that as it return after each if

Anyway current choice actually support Union

According to the model
```pyhon
    @property
    def choice(self):
        # type: () -> Union[port, flow, bgpv4, choice, choice, choice]
        """choice getter

        TBD

        Returns: Union[port, flow, bgpv4, choice, choice, choice]
        """
        return self._get_property('choice')
```